service/src/main/java/eu/europa/ec/fisheries/ers/service/bean/MovementModuleServiceBean.java

```diff
@@ -16,93 +16,40 @@
 
     @EJB
-    private MovementProducerBean movementProducer;
-
-    @EJB
-    private ActivityConsumerBean activityConsumer;
+    private MovementRestClient movementClient;
 
     /**
      * {@inheritDoc}
      */
     @Override
     @TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
-    public List<MovementType> getMovement(List<String> vesselIds, Date startDate, Date endDate) throws ServiceException {
-        try {
-            MovementQuery movementQuery = new MovementQuery();
-            addListCriteria(vesselIds, movementQuery);
-            addRangeCriteria(startDate, endDate, movementQuery);
-            movementQuery.setExcludeFirstAndLastSegment(true);
-
-            String request = MovementModuleRequestMapper.mapToGetMovementMapByQueryRequest(movementQuery);
-            String moduleMessage = movementProducer.sendModuleMessage(request, activityConsumer.getDestination());
-            TextMessage response = activityConsumer.getMessage(moduleMessage, TextMessage.class);
-            //TODO: Implement this call against new Movement Module!
-            /*
-            if (response != null && isNotUserFault(response)) {
-                List<MovementMapResponseType> mapResponseTypes = MovementModuleResponseMapper.mapToMovementMapResponse(response);
-                List<MovementType> movements = new ArrayList<>();
-                for (MovementMapResponseType movementMap : mapResponseTypes) {
-                    movements.addAll(movementMap.getMovements());
-                }
-                return movements;
-            } else {
-                throw new ServiceException("FAILED TO GET DATA FROM MOVEMENT");
-            }
-            */
-
-            return new ArrayList<>();
-
-        } catch (JMSException e) {
-            log.error("Exception in communication with movements", e);
-            throw new ServiceException(e.getMessage(), e);
-        }
-    }
-
-    private void addRangeCriteria(Date startDate, Date endDate, MovementQuery movementQuery) {
-        RangeCriteria rangeCriteria = new RangeCriteria();
-        rangeCriteria.setKey(RangeKeyType.DATE);
-        rangeCriteria.setFrom(DateFormatUtils.format(DateUtils.addDays(startDate, -1), "yyyy-MM-dd HH:mm:ss Z"));
-        rangeCriteria.setTo(DateFormatUtils.format(DateUtils.addDays(endDate, 1), "yyyy-MM-dd HH:mm:ss Z"));
-        movementQuery.getMovementRangeSearchCriteria().add(rangeCriteria);
-    }
-
-    private void addListCriteria(List<String> vesselIds, MovementQuery movementQuery) {
-        for (String vesselId : vesselIds) {
-            ListCriteria listCriteria = new ListCriteria();
-            listCriteria.setKey(SearchKey.CONNECT_ID);
-            listCriteria.setValue(vesselId);
-            movementQuery.getMovementSearchCriteria().add(listCriteria);
-        }
+    public List<MicroMovement> getMovement(List<String> vesselIds, Date startDate, Date endDate) throws ServiceException {
```

Comment:
Take Instants as paremeters directly? Or maybe I'll get around  to it in my never ending "update time handling" JIRA :)

Reply:
Yeah, I've tried to go with "less refactoring more get it to work" now otherwise it's easy to get bogged down.